root/components/Aliases/ArtistCreditList.js

```diff
@@ -17,7 +17,7 @@ import loopParity from '../../utility/loopParity';
 
 type Props = {
   +$c: CatalystContextT,
-  +artistCredits: $ReadOnlyArray<ArtistCreditT>,
+  +artistCredits: $ReadOnlyArray<{+id: number} & ArtistCreditT>,
```

Comment:
Wouldn't it make sense for this to be its own type?

Reply:
Or is this just saying "in this case, id is not optional"?

Reply:
Yeah, it's to say the `id` is always present here (otherwise we run into errors below that `id` can be `undefined`, which is obviously not true on this page).

Right now the `ArtistCreditT` definition is a bit weird since `id`, `entityType`, and `editsPending` are optional, even though they're not for ACs from the server. So we should probably have a separate `WritableArtistCreditT` (not sure of a good name) for ones which just have `names` (for use on edit pages).

Reply:
More of not sure good names: `ArtistCreditStubT`, `CoreArtistCreditT`.

Reply:
`ArtistCreditStubT` sounds pretty good. I was also thinking about `EditableArtistCreditT` since it's, well, intended to be editable and used on edit pages.

I wasn't planning on making these changes in this PR though.